torchvision/transforms/functional_tensor.py

```diff
@@ -18,6 +18,88 @@ def _get_image_size(img: Tensor) -> List[int]:
     raise TypeError("Unexpected type {}".format(type(img)))
 
 
+# torch.iinfo isn't scriptable so using this helper function
+# https://github.com/pytorch/pytorch/issues/41492
+def _max_value(dtype: int) -> int:
+    a = torch.tensor(2, dtype=dtype)
+    signed = 1 if torch.tensor(0, dtype=dtype).is_signed() else 0
+    bits = 1
+    max_value = torch.tensor(-signed, dtype=torch.long)
+    while(True):
+        next_value = a.pow(bits - signed).sub(1)
+        if next_value > max_value:
+            max_value = next_value
+            bits *= 2
+        else:
+            return max_value.item()
+    return max_value.item()
+
+
+def convert_image_dtype(image: torch.Tensor, dtype: int = torch.float) -> torch.Tensor:
```

Comment:
Can you move this function back to `functional.py`? My thinking about `functional_pil` and `functional_tensor` is that those are implementation details, and the user should only worry about `functional.py`.

Reply:
The issue that raises is then the implementation of `adjust_gamma` in functional_tensor (and any other functional impl's in which we'd like to use this) would need to depend on functional.py, but jit doesn't like that import.

Another option could be to have a `convert_image_dtype` in `functional.py` that takes either a PIL or Tensor, and delegates to this implementation as do other transforms. That doesn't seem to make sense to me unless there's also an equivalent PIL-based convert_image_dtype transform we would want to implement/use somewhere. To me this is seeming like a tensor-specific helper transform.

Reply:
Good point about the cyclic import. I don't think we want an implementation for PIL, but here is one proposal: add in `functional.py` something like

```python
# import to main namespace
# this is temporary until we merge the implementation of
# F_t inside functional.py
convert_image_dtype = F_t.convert_image_dtype
```

Reply:
If used defined `convert_image_dtype = F_t.convert_image_dtype` inside `functional.py`, then docstring for the method is not rendered. I moved back the implementation with the only dispatch to `F_t`.